packages/blockchain-wallet-v4-frontend/src/data/analytics/reducers.js

```diff
@@ -0,0 +1,18 @@
+import * as AT from './actionTypes'
+import { insert } from 'ramda'
+
+const INITIAL_STATE = []
+
+const analytics = (state = INITIAL_STATE, action) => {
+  const { type, payload } = action
+
+  switch (type) {
+    case AT.LOG_EVENT: {
+      return insert(0, payload, state)
```

Comment:
Can you add some error handling here

If event is not in ANALYTICS throw error
If payload is incorrect (not an array, check length?)

Another way would be to construct the payload yourself, similar to ```const createLog = (msgType, payload) => {
    return {
      type: msgType,
      file: payload.file,
      method: payload.method,
      timestamp: Date.now(),
      message: JSON.stringify(payload.message)
    }
  }```

Reply:
I thought `assert` worked well here. Really the two most likely failures are empty data and data being passed that is inconsistent with our "schema". In the future we can also check the event `category` so we're not creating duplicates by doing things like passing `send` versus `Send`